Configurable versions for gradle/kotlin/android plugin

jmfayard opened the pull request:
Hello, I wanted to fix indentation & update to Kotlin 1.1.60 ; android.gradle 3.0.1
In the meantime, I fixed the unit tests and clarified the main class

---

Update: the PR adds three parameters to the config

```
{
  "gradleVersion": "4.3.1",
  "androidGradlePluginVersion": "3.0.1",
  "kotlinVersion": "1.1.60",
}
```

jmfayard commented on src/main/kotlin/com/google/androidstudiopoet/BuildGradle.kt:
```diff
@@ -38,9 +38,6 @@ object BuildGradle {
             "buildscript {\n" +
-            (if (useKotlin)
-                "    ext.kotlin_version = '1.1.51'\n"
-            else "")+
```
removed it because it's already defined in the top build.gradle
no need to update it to a wrong version

NikitaKozlov commented on SampleConfig.json:
```diff
@@ -1,6 +1,8 @@
 {
   "projectName": "gennyCLI",
   "root": "./modules/",
+  "androidGradleVersion": "3.0.1",
```
please name it "androidGradlePluginVersion"

NikitaKozlov commented on src/main/assets/build.gradle:
```diff
@@ -7,7 +7,7 @@ buildscript {
     }
     dependencies {
-        classpath 'com.android.tools.build:gradle:3.0.0'
+        classpath 'com.android.tools.build:gradle:3.0.1'
```
Problem with customizing AGP version it that it is hardcoded here.

jmfayard replied:
is this file used? 
if I set androidGradlePluginVersion to 3.0.0 I have only one reference as expected

```
$ egrep -ir 'com.android.tools.build:gradle' modules/gennyCLI
./build.gradle:        classpath 'com.android.tools.build:gradle:3.0.0'
```

srmurguia replied:
This file was copied directly to the project folder, but after https://github.com/borisf/android-studio-poet/commit/b8d53687029d53e76dbb49801fcd7f440d54bbec#diff-469f26b37acf957d7e037aa3fc78cd01 this file is no longer used, I should have removed it on the same commit.

jmfayard replied:
I removed it

Commit:
Versions configurable & refactoring

New configurations available
  "gradleVersion": "4.3.1",
  "androidGradlePluginVersion": "3.0.1",
  "kotlinVersion": "1.1.60",

Update README

Refactor main class
